src/gt4py/next/program_processors/runners/dace_fieldview/gtir_sdfg.py

```diff
@@ -575,17 +611,57 @@ def visit_Lambda(
             (str(param.id), arg) for param, arg in zip(node.params, args, strict=True)
         ]
 
+        def _flatten_tuples(
+            name: str,
+            arg: gtir_builtin_translators.FieldopResult,
+        ) -> list[tuple[str, gtir_builtin_translators.FieldopData]]:
+            if isinstance(arg, tuple):
+                tuple_type = _get_tuple_type(arg)
+                tuple_field_names = [
+                    arg_name for arg_name, _ in dace_gtir_utils.get_tuple_fields(name, tuple_type)
+                ]
+                tuple_args = zip(tuple_field_names, arg, strict=True)
+                return list(
+                    itertools.chain(*[_flatten_tuples(fname, farg) for fname, farg in tuple_args])
+                )
+            else:
+                return [(name, arg)]
+
+        lambda_arg_nodes = dict(
+            itertools.chain(*[_flatten_tuples(pname, arg) for pname, arg in lambda_args_mapping])
+        )
+
         # inherit symbols from parent scope but eventually override with local symbols
         lambda_symbols = {
             sym: self.global_symbols[sym]
             for sym in symbol_ref_utils.collect_symbol_refs(node.expr, self.global_symbols.keys())
         } | {
-            pname: dace_gtir_utils.get_tuple_type(arg) if isinstance(arg, tuple) else arg.gt_type
+            pname: _get_tuple_type(arg) if isinstance(arg, tuple) else arg.gt_type
             for pname, arg in lambda_args_mapping
         }
 
+        # populate mapping from field name to domain offset
+        lambda_field_offsets: dict[str, Optional[list[dace.symbolic.SymExpr]]] = {}
+        for p_name, p_type in lambda_symbols.items():
+            if isinstance(p_type, ts.FieldType):
+                if p_name in lambda_arg_nodes:
+                    arg = lambda_arg_nodes[p_name]
+                    assert isinstance(arg, gtir_builtin_translators.FieldopData)
+                    lambda_field_offsets[p_name] = arg.offset
+                else:
+                    lambda_field_offsets[p_name] = self.field_offsets[p_name]
+            elif isinstance(p_type, ts.TupleType):
+                p_fields = dace_gtir_utils.get_tuple_fields(p_name, p_type, flatten=True)
+                lambda_field_offsets |= {
+                    p_tname: lambda_arg_nodes[p_tname].offset
+                    if p_tname in lambda_arg_nodes
+                    else self.field_offsets[p_tname]
+                    for p_tname, p_ttype in p_fields
+                    if isinstance(p_ttype, ts.FieldType)
```

Comment:
OPTIONAL: I am not sure, but I think it is somehow simpler to read this way:
```suggestion
                    (
                        p_tname: lambda_arg_nodes[p_tname].offset
                        if p_tname in lambda_arg_nodes
                        else self.field_offsets[p_tname]
                    )
                    for p_tname, p_ttype in p_fields
                    if isinstance(p_ttype, ts.FieldType)
```